sequential/backend/seq2pat.cpp

```diff
@@ -30,49 +30,67 @@ namespace patterns {
+                      &(this->uavri), &(this->lavri), // Note that the order is other way around
+                      &(this->umedi), &(this->lmedi),
+                      &(this->lgap), &(this->ugap),
+                      &(this->lavr), &(this->uavr),
+                      &(this->lspn),
+                      &(this->lmed), &(this->umed),
+                      &(this->num_minmax),
+                      &(this->num_avr),
+                      &(this->num_med),
+                      &(this->tot_gap),
+                      &(this->tot_spn),
+                      &(this->tot_avr),
+                      this->M, this->N, this->L,
+                      this->num_att, &(this->max_attrs), &(this->min_attrs),
+                      &(this->items),
+                      &(this->attrs));
         }
         catch(exception& e){
             throw e;
         }
 
         try{
             // Run frequent mining
-            return Freq_miner(  &mdd_q,
-                                &(this->uspni), &(this->lspni),
-                                &(this->uavri), &(this->lavri),
-                                &(this->umedi), &(this->lmedi),
-                                &(this->lavr), &(this->uavr),
-                                &(this->lspn), &(this->uspn),
-                                &(this->lmed), &(this->umed),
-                                &(this->num_minmax),
-                                &(this->num_avr),
-                                &(this->num_med),
-                                &(this->tot_spn),
-                                &(this->tot_avr),
-                                this->theta,
-                                this->L);
+            results = Freq_miner(mdd_q,
+                                 &(this->uspni), &(this->lspni),
+                                 &(this->uavri), &(this->lavri),
+                                 &(this->umedi), &(this->lmedi),
+                                 &(this->lavr), &(this->uavr),
+                                 &(this->lspn), &(this->uspn),
+                                 &(this->lmed), &(this->umed),
+                                 &(this->num_minmax),
+                                 &(this->num_avr),
+                                 &(this->num_med),
+                                 &(this->tot_spn),
+                                 &(this->tot_avr),
+                                 this->theta,
+                                 this->L);
+
+            // Delete MDD nodes
+            for (int i=0; i < (*datab_MDD).size(); i++){
+                if ((*datab_MDD)[i]!=NULL)
+                    (*datab_MDD)[i]->~Node();
+            }
+            // Delete vectors defined in freq_miner.cpp
+            std::vector<bool>().swap(indic_vec);
+	        std::vector<vector<int>>().swap(result);
+	        // Delete pointers
+            delete datab_MDD;
+            // mdd_q, the queue should be empty after calling Freq_miner()
```

Comment:
This could be referring to the backend implementation in [freq_miner](https://github.com/fidelity/seq2pat/blob/create_one_hot_embedding/sequential/backend/freq_miner.cpp#L38). I might not be able to explain the detailed steps, but one observation is that all the objects in this queue will be popped out after running the algorithm. Before the object pointer gets popped, the object gets destroyed manually by calling `~Pattern()`.